### PR TITLE
[PRO-138] Account page and resend confirmation functionality

### PR DESCRIPTION
**Pull request by Timmehs**
### Changes
* New /account page for logged-in users
* Moved "sign out" functionality to account page
* Added a notification to account page when user has not confirmed their email, with a "Resend" button to resend a confirmation link.
* Placeholder "Password change" section.
* Placeholder "Delete account" section.

### Screens
**Viewing account page with unconfirmed account**
<div style="display: flex; flex: row; width: 100%;">
image
image
</div>

**Post-confirmation**
image

### Other thoughts
* It turns out the StaticPage does represent a pattern that is used beyond just the "static content" pages. Maybe we should just reduce it to a more generic header component, e.g. `IconHeader`
* We're do for a standardized modal component, as these patterns are just getting duplicated.

### CODE REVIEW

File: src/types/User.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 type User = {
   email: string
-  confirmed: boolean
+  isConfirmed: boolean
+  confirmationSentAt?: string
```
Comment (Timmehs) [inline src/types/User.ts:4]: This adjustment is for API changes found in this PR: https://github.com/ProjectProtocol/project-protocol-api/pull/104

File: src/pages/Account.tsx [hunk 2]
```diff
@@ -1,15 +1,101 @@
...
-export default function Account() {
-  const { user } = useAuth()
+export default function AccountView() {
+  const { user, handleLogout } = useAuth()
+  const [resentCode, setResentCode] = useState(false)
+  const [showDeleteModal, setShowDeleteModal] = useState(false)
 
-  if (!user) return <Navigate to="/" />
+  if (!user) {
+    console.log('what the fuck')
```
Comment (eddie-m-m) [inline src/pages/Account.tsx:18]: We can commit this? (lol)

**Review by eddie-m-m — COMMENTED** [review]
I signed up a new user. Confirmed account by copy/pasting confirmation token in the browser. 
In Account Settings, it still is giving me the option to `resend code`. When I click on it, it toasts an error.

**Comment by Timmehs** [comment]
> I signed up a new user. Confirmed account by copy/pasting confirmation token in the browser. In Account Settings, it still is giving me the option to `resend code`. When I click on it, it toasts an error.

The react code is checking for `user.isConfirmed` but your version of the API sends back a user object with `user.confirmed`. I just merged the API update, so if you go to your API directory and pull the latest `main`, you should see the confirmation UI go away after a page refresh.

File: src/pages/Account.tsx (hunk 2 above)
Comment (Timmehs) [reply src/pages/Account.tsx]: Nothin to see here 🙄

**Comment by Timmehs** [comment]
@eddie-m-m Good catch on the console message. I added an eslint rule to help us catch those.

**Review by eddie-m-m — APPROVED** [review]
Click-through was successful. It included:

- [x] sign in existing user and sign out
- [x] sign up new user 
- [x] re-send code on `Account` page prior to confirmation
- [x] attempt to confirm with old token
- [x] confirm with new (current) token
- [x] sign out newly created user and sign back in 

Everything worked great. All expected UI updates and toasts function properly.

Great stuff!

**Comment by eddie-m-m** [comment]
RE: Other Thoughts

- I agree that we should slice out the `StaticPage` header pattern into a separate component. I'm wondering if we will eventually have other sorts of header patterns and so whether we should start with a base Header and extend from it, e.g. IconHeader right now, and whatever else down the line.

- A standardized modal component sounds good.

**Comment by Timmehs** [comment]
Tickets for modal and iconheader created 👍

**Pull request merged**